lib/Doctrine/DBAL/Connection.php

```diff
@@ -1264,6 +1264,8 @@ public function getWrappedConnection() : DriverConnection
     {
         $this->connect();
 
+        assert($this->_conn !== null);
```

Comment:
I'm not sure what your policy is in this case: `assert()`? `throw new LogicException()?`

There are several other places, as you'll see in the diff, where we know that the variable will not be null, but phpstan cannot dig deep enough to figure it out by itself.

Reply:
The assertions should be added only if the issue reported by PHPStan is in fact impossible but this is not obvious from the type system. On the other hand, if an exception is added, there should be a test case that covers it being thrown.

In this case, the assertion looks legit. While the API could use some improvement. There should be a method that always _returns_ a connection, not sets it on the object (out of scope).